itest/lnd_sweep_test.go

```diff
@@ -2150,3 +2152,114 @@ func runBumpFee(ht *lntest.HarnessTest, alice *node.HarnessNode) {
+
+	// Unwrap the results.
+	chanPoint := chanPoints[0]
+	alice := nodes[0]
+
+	// Alice force closes the channel which has no HTLCs at stake.
+	_, closingTxID := ht.CloseChannelAssertPending(alice, chanPoint, true)
+	require.NotNil(ht, closingTxID)
+
+	// Alice should see one waiting close channel.
+	ht.AssertNumWaitingClose(alice, 1)
+
+	// Alice should have 2 registered sweep inputs. The anchor of the local
+	// commitment tx and the anchor of the remote commitment tx.
+	ht.AssertNumPendingSweeps(alice, 2)
+
+	// Calculate the commitment tx fee rate.
+	closingTx := ht.AssertTxInMempool(closingTxID)
+	require.NotNil(ht, closingTx)
+
+	// The default commitment fee for anchor channels is capped at 2500
+	// sat/kw but there might be some inaccuracies because of the witness
+	// signature length therefore we calculate the exact value here.
+	closingFeeRate := ht.CalculateTxFeeRate(closingTx)
+	parentWeight := ht.CalculateTxWeight(closingTx)
+	parentFee := ht.CalculateTxFee(closingTx)
+
+	// Expect a 2 input 1 output transaction when bumping the force-close
+	// transaction. We have a large enough input because alice had a 1 BTC
+	// input to start with.
+	estimator := input.TxWeightEstimator{}
+	estimator.AddTaprootKeySpendInput(txscript.SigHashDefault)
+
+	// We are dealing with non-taproot channels here.
+	estimator.AddWitnessInput(input.AnchorWitnessSize)
+
+	// LND by default creates p2tr outputs.
+	estimator.AddP2TROutput()
+	childWeight := estimator.Weight()
+
+	// We increase the fee rate of the fee function by 100% to make sure
+	// we trigger a cpfp-transaction.
+	newFeeRate := closingFeeRate * 2
+
+	// We need to make sure that the budget can cover the fees for bumping
+	// the close transction taking also the parent fee into account.
+	// Because we are not taking the parent fee rate into account when
+	// crafting the initial cpfp transaction even less will be used but
+	// technically this is the minimum budget we need to bump the fee rate
+	// to the new starting fee rate of the fee function.
+	minBudget := newFeeRate.FeeForWeight(parentWeight+childWeight) -
+		parentFee
+
+	bumpFeeReq := &walletrpc.BumpForceCloseFeeRequest{
+		ChanPoint:   chanPoint,
+		SatPerVbyte: uint64(newFeeRate.FeePerVByte()),
+		// The budget needs to be high enough to pay for the fee because
+		// the anchor does not have an output value high enough to pay
+		// for itself.
+		Budget: uint64(minBudget),
+		// We use a force param to create the sweeping tx immediately.
+		Immediate: true,
+	}
+
+	alice.RPC.BumpForceCloseFee(bumpFeeReq)
+
+	// We expect the initial closing transaction and the local anchor cpfp
+	// transaction because alice force closed the channel.
+	// NOTE: We don't compare a feerate but only make sure that a cpfp
+	// transaction was triggered. The sweeper increases the fee rate
+	// periodically with every new incoming block and the selected fee
+	// function.
+	ht.AssertNumTxsInMempool(2)
```

Comment:
Not sure what you mean, but I calculated the weight of a child because there is still not child transaction in the mempool before bumping the close transaction ?

Reply:
yeah if we use an arbitrarily large enough budget then we can make the test setup a bit easier, then we don't need to care about weights here since we are not asserting an expected feerate was used during the process?

Reply:
just chose a large enough budget so we don't need to care about this.